src/amr/messengers/hybrid_hybrid_messenger_strategy.hpp

```diff
@@ -501,14 +556,23 @@ namespace amr
             for (auto& patch : level)
             {
                 auto dataOnPatch = resourcesManager_->setOnPatch(
-                    *patch, hybridModel.state.electromag, hybridModel.state.J, EM_old_, Jold_);
+                    *patch, hybridModel.state.electromag, hybridModel.state.J,
+                    hybridModel.state.ions, EM_old_, Jold_, NiOldUser_, ViOld_);
+
                 resourcesManager_->setTime(EM_old_, *patch, currentTime);
                 resourcesManager_->setTime(Jold_, *patch, currentTime);
+                resourcesManager_->setTime(NiOldUser_, *patch, currentTime);
+                resourcesManager_->setTime(ViOld_, *patch, currentTime);
```

Comment:
maybe overkill but this could be a tuple apply

something like

```
core::apply(std::forward_as_tuple(EM_old_, Jold_,NiOldUser_, ViOld_), [&](auto& item){
  resourcesManager_->setTime(item, *patch, currentTime);
});
```

Reply:
appealing but I think I prefer to keep it "simple"